### PR TITLE
Add cleanup time and last transition time to OktaAssignment.

### PR DESCRIPTION
**Pull request by mdwn**
The OktaAssignment now has a cleanup time that will be specified on creation and each action has a LastTransition time. The cleanup time specifies when the Okta service should clean up the Okta assignment, and the last transition time will be used to show when the action last transitioned.

It should be noted that cleanup time is maintained separately from the notion of expiry as we want Okta assignments to stick around until they're cleaned up, otherwise we have no record of what Teleport has done via the Okta API.

LastTransition time can be used to timeout actions that have been stuck processing for too long.

### CODE REVIEW

File: api/types/okta.go [hunk 1]
```diff
@@ -180,6 +181,8 @@ type OktaAssignment interface {
 	GetUser() string
 	// GetActions will return the list of actions that will be performed as part of this assignment.
 	GetActions() []OktaAssignmentAction
+	// GetCleanupTime will return the optional time that the assignment should be cleaned up.
+	GetCleanupTime() *time.Time
```
Comment (russjones) [inline api/types/okta.go:185]: Any reason this is a pointer? Are you using it being `nil` to differentiate states (not set, set as zero time, set at a particular time)?

Comment (mdwn) [reply api/types/okta.go:185]: This field is optional, so nil is being used to determine if it's not set.

Comment (r0mant) [reply api/types/okta.go:185]: Nit: technically you can make it non-pointer and use `IsZero()` to determine whether it's set or not but I don't have a strong opinion on this one.

**Review by r0mant — APPROVED** [review]

File: api/types/okta.go (hunk 1 above)
Comment (russjones) [reply api/types/okta.go:185]: Feel the same way as Roman on this one.

**Review by russjones — APPROVED** [review]

**Pull request merged**